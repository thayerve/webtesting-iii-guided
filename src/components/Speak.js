import React from 'react';

const Speak = props => {
    return (
        <>
        <button onClick={props.speak}>Speak</button>
        <div>{props.message}</div>
        </>
    )
};

export default Speak;