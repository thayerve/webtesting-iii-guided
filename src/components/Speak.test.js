import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import Speak from './Speak';

describe('<Speak />', () => {
    // 2. write this test
    it('matches snapshot', () => {
      const tree = renderer.create(<Speak />); // generates a DOM tree
  
      // snapshots are a JSON representation of the DOM tree
      expect(tree.toJSON()).toMatchSnapshot();
    });

    // implementation details that we don't care about and won't test:
    // - what props.speak is doing...
    // - if props.speak is invoked...

    // What we do want to test: 
    // - What does the user see in the app? Do they see the message when they click the button?

    it('displays a message when "speak" button is clicked', ()=>{
        const speak = jest.fn();
        // ^ this is a mock

        const { getByText } = render(<Speak speak={speak} />);

        const button = getByText(/speak/i);
        fireEvent.click(button);

        // ^ can also be written as:
        // fireEvent.click(getByText(/speak/i));
        
        expect(speak).toHaveBeenCalled();
        
    })

  });