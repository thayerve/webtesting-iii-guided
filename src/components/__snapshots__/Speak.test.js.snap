// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speak /> matches snapshot 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div />,
]
`;
